presto-function-namespace-managers/src/main/java/com/facebook/presto/functionNamespace/AbstractSqlInvokedFunctionNamespaceManager.java

```diff
@@ -176,11 +178,16 @@ public Optional<UserDefinedType> getUserDefinedType(QualifiedObjectName typeName
         try {
             return Optional.of(userDefinedTypes.getUnchecked(typeName));
         }
-        catch (PrestoException e) {
-            if (e.getErrorCode().equals(NOT_FOUND.toErrorCode())) {
-                return Optional.empty();
+        catch (UncheckedExecutionException e) {
+            Throwable cause = e.getCause();
+            if (cause instanceof PrestoException) {
+                PrestoException prestoException = (PrestoException) cause;
+                if (prestoException.getErrorCode().equals(NOT_FOUND.toErrorCode())) {
```

Comment:
why do we have a NOT_FOUND as error code?

Reply:
this gets thrown by the function namespace manager if the type isn't found.  Here I was trying to keep the behavior that the previous code was clearly trying to achieve, but wasn't doing correctly.